sdk/nodejs/provider.ts

```diff
@@ -39,35 +39,29 @@ export class Provider extends pulumi.ProviderResource {
-            inputs["endpoints"] = pulumi.output(args ? args.endpoints : undefined).apply(JSON.stringify);
-            inputs["forbiddenAccountIds"] = pulumi.output(args ? args.forbiddenAccountIds : undefined).apply(JSON.stringify);
-            inputs["insecure"] = pulumi.output(args ? args.insecure : undefined).apply(JSON.stringify);
-            inputs["maxRetries"] = pulumi.output(args ? args.maxRetries : undefined).apply(JSON.stringify);
-            inputs["profile"] = (args ? args.profile : undefined) || utilities.getEnv("AWS_PROFILE");
-            inputs["region"] = (args ? args.region : undefined) || utilities.getEnv("AWS_REGION", "AWS_DEFAULT_REGION");
-            inputs["s3ForcePathStyle"] = pulumi.output(args ? args.s3ForcePathStyle : undefined).apply(JSON.stringify);
-            inputs["secretKey"] = args ? args.secretKey : undefined;
-            inputs["sharedCredentialsFile"] = args ? args.sharedCredentialsFile : undefined;
-            inputs["skipCredentialsValidation"] = pulumi.output(args ? args.skipCredentialsValidation : undefined).apply(JSON.stringify);
-            inputs["skipGetEc2Platforms"] = pulumi.output(args ? args.skipGetEc2Platforms : undefined).apply(JSON.stringify);
-            inputs["skipMetadataApiCheck"] = pulumi.output(args ? args.skipMetadataApiCheck : undefined).apply(JSON.stringify);
-            inputs["skipRegionValidation"] = pulumi.output(args ? args.skipRegionValidation : undefined).apply(JSON.stringify);
-            inputs["skipRequestingAccountId"] = pulumi.output(args ? args.skipRequestingAccountId : undefined).apply(JSON.stringify);
-            inputs["token"] = args ? args.token : undefined;
-        }
-        if (!opts) {
-            opts = {}
-        }
-
-        if (!opts.version) {
-            opts.version = utilities.getVersion();
+            inputs.accessKey = args.accessKey;
+            inputs.allowedAccountIds = args.allowedAccountIds;
+            inputs.assumeRole = args.assumeRole;
+            inputs.endpoints = args.endpoints;
+            inputs.forbiddenAccountIds = args.forbiddenAccountIds;
+            inputs.insecure = args.insecure;
+            inputs.maxRetries = args.maxRetries;
+            inputs.profile = args.profile || utilities.getEnv("AWS_PROFILE");
+            inputs.region = args.region || utilities.getEnv("AWS_REGION", "AWS_DEFAULT_REGION");
+            inputs.s3ForcePathStyle = args.s3ForcePathStyle;
+            inputs.secretKey = args.secretKey;
+            inputs.sharedCredentialsFile = args.sharedCredentialsFile;
+            inputs.skipCredentialsValidation = args.skipCredentialsValidation;
+            inputs.skipGetEc2Platforms = args.skipGetEc2Platforms;
+            inputs.skipMetadataApiCheck = args.skipMetadataApiCheck;
+            inputs.skipRegionValidation = args.skipRegionValidation;
+            inputs.skipRequestingAccountId = args.skipRequestingAccountId;
+            inputs.token = args.token;
```

Comment:
this is the relevant change.  we just copy the values over untouched (except for values that need to do utilities.getEnv to get defaults).